src/Shouldly/Configuration/FirstNonShouldlyMethodFinder.cs

```diff
@@ -20,20 +20,23 @@ public class FirstNonShouldlyMethodFinder : ITestMethodFinder
 
         public TestMethodInfo GetTestMethodInfo(StackTrace stackTrace, int startAt = 0)
         {
-            var i = startAt;
-            StackFrame callingFrame;
-            do
+            for (var i = startAt; stackTrace.GetFrame(i) is { } frame; i++)
             {
-                callingFrame = stackTrace.GetFrame(i++);
-            } while (callingFrame.GetMethod().IsShouldlyMethod() || IsCompilerGenerated(callingFrame.GetMethod()));
+                if (frame.GetMethod() is { } method && !method.IsShouldlyMethod() && !IsCompilerGenerated(method))
```

Comment:
`frame.GetMethod() is { } method`  what does that syntax mean?

Reply:
It means if `frame.GetMethod()` is not null, put it in the `method` variable and evaluate true. Otherwise evaluate false.

It's a degenerate case of C# 8 property patterns. A non-degenerate case would actually have at least one property:

`if (frame.GetMethod() is { Name: "Foo" } method)`

Reply:
If you like, I'll rewrite as:

```cs
var method = frame.GetMethod();
if (method != null && !method.IsShouldlyMethod() && !IsCompilerGenerated(method))
```

Reply:
not at all. i am enjoying the education

Reply:
Oh, great! 😀